Start an Application object

whit537 opened the pull request:
This creates a new `Application` object that subsumes `Website` and should eventually eat all of `wireup`(!). ~~Immediate next step is to make an `Email` mixin for it: #4346.~~ We need to make this available to payday and models before we can work on email: #4350.

### Todo

- [x] change base to `master`

whit537 commented on www/~/%username/emails/verify.html.spt:
```diff
@@ -55,33 +56,34 @@ suppress_sidebar = True
+                        ("<b>%s</b>"|safe) % email_address) }}</p>
             {% endif %}
-            <p data-email="{{ email }}"><button class="resend">{{ _("Resend verification email") }}</button></p>
+            <p>Result: {{ result }}</p>
```
Stray log line.

whit537 commented:
Rebased, was 9d154116aa75093d148a724acf2c7e00eef2f0a3.

whit537 commented:
Okay! This PR is weighing in at +/-440 ... not as bad as I thought. :)

whit537 commented:
Erm, well, now up to 516. 😁

whit537 commented:
Rebased, was 24c1eb672e566fae56db1ca19ffe82f9277d53f4.

whit537 commented:
This one is up next, @nobodxbodon et al.

whit537 commented:
"cli" means "command line interface"

[Google search](https://www.google.com/search?q=cli) suggests that this is a common usage.

nobodxbodon commented:
@whit537 thanks. I didn't realize the dequeue_emails/list_email_queue are command line tools. Now it makes sense. Would you like it to be reviewed more in depth by someone else or shall I merge it?

whit537 commented:
Yeah, we added those scripts a month or two ago when we wanted to send out a notification about our new terms of service (#4232).

Rebased again (was d39c8a36bbd5a82ebb7396569065e6996cd290b3) to catch up with recent [bugfix](http://gratipay.slackarchive.io/gratipay/-/1488309786.007464/1488482054.007634/1488480110007570/) [deploys](http://gratipay.slackarchive.io/gratipay/-/1488309786.007464/1488482054.007634/1488481161007633/). I'm ready to go here if you are, @nobodxbodon. Merge away! :-)

nobodxbodon commented:
I wonder how to restart travis manually? Or is it possible to config travis to retry automatically in case of timeout (no failed cases)?

mattbk commented:
If you don't have whatever special privileges @whit537 has, one method is to make a small commit and push to jumpstart Travis again.

whit537 commented:
Yeah, there's a little "Restart build" button at upper right. Not sure why you don't have perms ...

whit537 commented:
![screen shot 2017-03-03 at 7 43 20 am](https://cloud.githubusercontent.com/assets/134455/23551530/20f94d5c-ffe5-11e6-97e6-d3a96b123d1c.png)

whit537 commented:
Restarted ...

whit537 commented:
Green! :-)

mattbk commented:
> Yeah, there's a little "Restart build" button at upper right. Not sure why you don't have perms ...

On Travis itself? I don't have that.

whit537 commented:
After poking around for a bit in the Travis UI, the closest thing I found to something that might help us here is the "Sync" button for the Gratipay GitHub org. I've gone ahead and clicked that ... did that update perms sufficient to enable the "Restart build" button for yinz?

![screen shot 2017-03-03 at 11 17 34 am](https://cloud.githubusercontent.com/assets/134455/23558835/093a3d34-0003-11e7-90f9-d7085cda9547.png)

mattbk commented:
Got it. I had to be signed into Travis using GitHub credentials.

whit537 commented:
Ah! That makes sense. :-)

!m @mattbk

whit537 commented:
💃

whit537 commented:
This is confusing. I'm trying to deploy this and getting an error:

```
[gratipay] $ ./deploy.sh 
Already up-to-date.
Traceback (most recent call last):
  File "gratipay/wireup.py", line 11, in <module>
    import aspen
  File "/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/aspen/__init__.py", line 65, in <module>
    import pkg_resources
  File "/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 37, in <module>
    import email.parser
  File "/Users/whit537/personal/gratipay/gratipay.com/gratipay/email.py", line 5, in <module>
    from aspen.simplates.pagination import parse_specline, split_and_escape
  File "/Users/whit537/personal/gratipay/gratipay.com/env/lib/python2.7/site-packages/aspen/simplates/__init__.py", line 15, in <module>
    from .. import log
ImportError: cannot import name log
[gratipay] $
```

whit537 commented:
```
[gratipay] $ git log | head -n1
commit 796d36e914a44b88e60964416e754c0b473ca163
[gratipay] $
```

whit537 commented:
I ran `make clean env`.

whit537 commented:
796d36e914a44b88e60964416e754c0b473ca163 is the merge of this PR into `master`:

https://github.com/gratipay/gratipay.com/commits/796d36e914a44b88e60964416e754c0b473ca163

whit537 commented:
Moving to #4364.
